packages/tailwind/src/tailwind.tsx

```diff
@@ -26,156 +27,93 @@ export interface TailwindProps {
+      mediaQueryClassesForAllElement =
+        mediaQueryClassesForAllElement.concat(nonInlinableClasses);
+      nonInlineStylesRootToApply.append(nonInlineStyleNodes);
+
+      if (nonInlinableClasses.length > 0 && !hasNonInlineStylesToApply) {
+        hasNonInlineStylesToApply = true;
       }
-    }
 
-    // If the element is a component, it's best to only process the children
-    // after the component deals with them. The purpose of this is so that
-    // advanced element manipulation can still work properly.
-    if (element.props.children && !isComponent(element)) {
-      const processedChillren = React.Children.map(
-        element.props.children,
-        (child) => {
-          if (React.isValidElement<EmailElementProps>(child)) {
-            return processElement(child);
-          }
-
-          return child;
-        },
-      );
-      propsToOverwrite.children =
-        processedChillren && processedChillren.length === 1
-          ? processedChillren[0]
-          : processedChillren;
+      return elementWithInlinedStyles;
     }
 
-    if (element.props.className) {
-      const { styles, residualClassName } = inline(element.props.className);
-      propsToOverwrite.style = {
-        ...styles,
-        ...element.props.style,
-      };
-      if (!isComponent(element)) {
-        if (residualClassName.trim().length > 0) {
-          propsToOverwrite.className = residualClassName;
-          /*
-          We sanitize only the class names of Tailwind classes that we are not going to inline
-          to avoid unpredictable behavior on the user's code. If we did sanitize all class names
-          a user-defined class could end up also being sanitized which would lead to unexpected
-          behavior and bugs that are hard to track.
-        */
-          for (const singleClass of nonInlinableClasses) {
-            propsToOverwrite.className = propsToOverwrite.className.replace(
-              singleClass,
-              sanitizeClassName(singleClass),
-            );
-          }
-        } else {
-          propsToOverwrite.className = undefined;
-        }
-      }
-    }
+    return node;
+  });
 
-    const newProps = {
-      ...element.props,
-      ...propsToOverwrite,
-    };
-    const newChildren = propsToOverwrite.children
-      ? propsToOverwrite.children
-      : element.props.children;
-
-    if (isComponent(element)) {
-      const component =
-        typeof element.type === "object"
-          ? // @ts-expect-error - we know this is a component with a render function
-            (element.type.render as React.FC)
-          : (element.type as React.FC);
-      const renderedComponent = component({
-        ...element.props,
-        ...propsToOverwrite,
-      });
-
-      if (React.isValidElement<EmailElementProps>(renderedComponent)) {
-        return processElement(renderedComponent);
+  if (hasNonInlineStylesToApply) {
+    let hasAppliedNonInlineStyles = false as boolean;
```

Comment:
```suggestion
    let hasAppliedNonInlineStyles = false;
```
is the casting really needed?